types/tx_msg.go

```diff
@@ -36,21 +41,27 @@ type (
 		GetSignature() []byte
 	}
 
-	// Tx defines the interface a transaction must fulfill.
-	Tx interface {
-		HasValidateBasic
-
+	// HasMsgs defines an interface a transaction must fulfill.
+	HasMsgs interface {
 		// GetMsgs gets the all the transaction's messages.
 		GetMsgs() []Msg
 	}
 
+	// Tx defines an interface a transaction must fulfill.
+	Tx interface {
+		HasMsgs
+
+		// GetMsgsV2 gets the transaction's messages as google.golang.org/protobuf/proto.Message's.
+		GetMsgsV2() ([]protov2.Message, error)
+	}
+
 	// FeeTx defines the interface to be implemented by Tx to use the FeeDecorators
 	FeeTx interface {
 		Tx
 		GetGas() uint64
 		GetFee() Coins
-		FeePayer() AccAddress
-		FeeGranter() AccAddress
+		FeePayer() []byte
+		FeeGranter() string
```

Comment:
Confused by this.  It looks like we're using []byte in place of AccAddress in most places but here we're using both []byte and string as return types.  Shouldn't they be both []byte?

Reply:
Yeah, just more refactoring that I wanted to keep out of this already very large PR

Reply:
So we should make an issue to refactor these both to []byte?

Reply:
Okay to create a separate issue to address this?